nsv13/code/__HELPERS/overmap.dm

```diff
@@ -1,11 +1,15 @@
-/atom/proc/get_overmap() //Helper proc to get the overmap ship representing a given area.
+/atom/proc/get_overmap(failsafe = FALSE) //Helper proc to get the overmap ship representing a given area.
 	RETURN_TYPE(/obj/structure/overmap)
 	if(isovermap(loc))
 		return loc
 	if(!z) // We're in something's contents
 		if(!loc) // Or not...
 			return FALSE
 		return loc.get_overmap() // Begin recursion!
+	if(!SSmapping.z_list) //Being called by pre-mapping init call
+		if(!failsafe) //Only retry once
+			addtimer(CALLBACK(src, .proc/get_overmap, TRUE), 30 SECONDS)
+		return FALSE
```

Comment:
I can only support this one proc getting more absurdly complex purely to handle all the edge cases